_posts/2025-04-28-sparse-autodiff.md

```diff
@@ -749,14 +749,34 @@ This backup storage enables the use of **fewer distinct colors**, reducing the c
 Powell and Toint <d-cite key="powellEstimationSparseHessian1979"></d-cite> were the first to notice symmetry-related optimizations, before Coleman and Moré <d-cite key="colemanEstimationSparseHessian1984"></d-cite> made the connection to graph coloring explicit.
 While symmetric coloring and decompression are more computationally expensive than their nonsymmetric counterparts, this cost is typically negligible compared to the savings we get from fewer HVPs.
 
+## Applications
+
+ASD is useful in applications which involve the computation of full Jacobian or Hessian matrices with sparsity.
+Let us discuss one of the examples given in the 2024 ICLR blog post *How to compute Hessian-vector products?* <d-cite key="dagreouHowComputeHessianvector2024"></d-cite>: Newton's method.
+This canonical algorithm for nonlinear optimization uses Hessians within local steps of the form $[\nabla^2 f(\mathbf{x})]^{-1} \nabla f(\mathbf{x})$.
+The inverse $[\nabla^2f(\mathbf{x})]^{-1}$ is never created explicitly, since it suffices to solve a linear system $\nabla^2f(\mathbf{x}) \cdot \mathbf{v} = \nabla f(\mathbf{x})$.
+As the blog post argues, for such systems, matrix-free iterative solvers combine well with Hessian operators because they only require successive evaluations of HVPs.
+The main advantage of iterative solvers is their small memory footprint and their ease of implementation.
+
+However, when it is possible to materialize the whole Hessian matrix, direct linear solvers can be used (e.g. based on matrix factorization).
+As we have seen, ASD unlocks this option whenever the Hessian matrix is sparse enough.
+In terms of performance, computing a sparse Hessian matrix might require fewer HVPs overall than applying an iterative solver like the conjugate gradient method.
+The exact performance depends on the number of colors in the sparsity pattern of the Hessian, as well as the numerical precision expected from the iterative solver (which influences the number of iterations, hence the number of HVPs).
+In terms of numerical accuracy, direct solvers are more robust than their iterative counterparts.
+Finally, in terms of compatibility, some prominent nonlinear optimization libraries only support Hessian matrices, and not Hessian operators.
```

Comment:
This would be a stronger argument if we named (and cited) at least one,

Reply:
Honestly I would put it almost at the top of the blog.
We can name the libraries @amontoison mentioned but we have to double-check in the docs first

Reply:
We can also move the entire applications section, but I'm mostly referring to the last sentence here. 
Performance and numerical stability are more important than compatibility with some software most ML folks will never have heard of. 
Maybe less is more and we should just remove the last sentence entirely.